### PR TITLE
feat: add MovieController

### PR DESCRIPTION
**Pull request by DFraczek**
# MovieController

This controller handles the HTTP requests for managing `MovieDisc` entities in the system. It provides several endpoints to perform CRUD (Create, Read, Update, Delete) operations and search for movies by various criteria.

## Endpoints

   - **POST** `/api/movie` (Create a new Movie)
   - **GET** `/api/movie/{id}` (Get movie by ID)
   - **GET** `/api/movie/director/{director}` (Get movies by director)
   - **GET** `/api/movie/genre/{genre}` (Get movies by genre)
   - **GET** `/api/movie/fileFormat/{fileFormat}` (Get movies by file format)
   - **GET** `/api/movie/resolution/{resolution}` (Get movies by resolution)
   - **GET** `/api/movie/duration/{minDuration}/{maxDuration}` (Get movies by duration range)
   - **GET** `/api/movie/titleAndDirector/{title}/{director}` (Get movies by title and director)
   - **PUT** `/api/movie/{id}` (Update a movie)
   - **DELETE** `/api/movie/{id}` (Delete a movie)

## Summary

- This controller manages `MovieDisc` entities by offering endpoints for creating, reading, updating, and deleting movies in the system.
- It allows searching for movies based on various filters like director, genre, file format, resolution, and duration.
- The `titleAndDirector` endpoint specifically helps find movies using both the title and director name.
- It responds with appropriate HTTP status codes and messages for success, failure, and not found scenarios.

## Key Changes

- **GET Endpoints**: Added several endpoints to filter movies by `director`, `genre`, `fileFormat`, `resolution`, and `duration`.
- **Search Functionality**: Introduced an endpoint to find movies based on both `title` and `director`.
- **CRUD Support**: Ensured that the controller supports full CRUD operations for the `MovieDisc` entity.
- **PathVariable and RequestParam**: Used `PathVariable` for resource IDs and specific attributes like genre, file format, etc., and `RequestParam` for query parameters (like duration range and title-director combination).

### CODE REVIEW

File: library/src/main/java/org/pollub/library/item/controller/MovieController.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public ResponseEntity<List<MovieDisc>> getMoviesByFileFormat(@PathVariable String fileFormat) {
+        return ResponseEntity.ok(movieService.findByFileFormat(fileFormat));
+    }
+
+    @GetMapping("/resolution/{resolution}")
+    public ResponseEntity<List<MovieDisc>> getMoviesByResolution(@PathVariable String resolution) {
+        return ResponseEntity.ok(movieService.findByResolution(resolution));
+    }
+
+    @GetMapping("/duration/{minDuration}/{maxDuration}")
```
Comment (blackspyek) [inline library/src/main/java/org/pollub/library/item/controller/MovieController.java]: If you use @RequestParam then its gonna be like this
http://localhost:8080/api/movie?title=abc&director=asdad
not like you did /abc/asdds thats called @PathVariable
so remove /{title}/{director}

File: library/src/main/java/org/pollub/library/item/controller/MovieController.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public ResponseEntity<List<MovieDisc>> getMoviesByDurationBetween(
+            @RequestParam Integer minDuration, @RequestParam Integer maxDuration) {
+        return ResponseEntity.ok(movieService.findByDurationBetween(minDuration, maxDuration));
+    }
+
+    @GetMapping("/titleAndDirector/{title}/{director}")
```
Comment (blackspyek) [inline library/src/main/java/org/pollub/library/item/controller/MovieController.java]: like above

**Review by blackspyek — CHANGES_REQUESTED** [review]
Fix commented

File: library/src/main/java/org/pollub/library/item/controller/MovieController.java (hunk 2 above)
Comment (DFraczek) [reply library/src/main/java/org/pollub/library/item/controller/MovieController.java]: fixed

File: library/src/main/java/org/pollub/library/item/controller/MovieController.java (hunk 1 above)
Comment (DFraczek) [reply library/src/main/java/org/pollub/library/item/controller/MovieController.java]: fixed

**Review by blackspyek — APPROVED** [review]
LGTM

**Pull request merged**